src/util/gcalUtils.ts

```diff
@@ -0,0 +1,236 @@
+import moment, { Moment } from 'moment';
+import { rrulestr } from 'rrule';
+import {
+  CalendarEvent,
+  CancelledCalendarEvent,
+  ChangedCalendarEvent,
+  GCalEvent,
+  GCalResponse,
+} from '../types';
+
+export async function loadGAPI(apiKey: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const script = document.createElement('script');
+    script.src = 'https://apis.google.com/js/api.js';
+    document.body.appendChild(script);
+    script.onload = () => {
+      gapi.load('client', () => {
+        gapi.client.init({ apiKey }).then(() => {
+          gapi.client
+            .load(
+              'https://content.googleapis.com/discovery/v1/apis/calendar/v3/rest'
+            )
+            .then(
+              () => resolve(),
+              (err: unknown) => reject(err)
+            );
+        });
+      });
+    };
+  });
+}
+
+export async function getEvents(
+  calendarId: string,
+  maxResults = 1000
+): Promise<GCalResponse> {
+  return gapi.client.calendar.events.list({
+    calendarId,
+    maxResults,
+  });
+}
+
+export function isMultiEvent(startTime: Moment, endTime: Moment): boolean {
+  return (
+    moment.duration(endTime.diff(startTime)).asHours() >= 24 ||
+    (!startTime.isSame(endTime, 'day') && endTime.hour() >= 12)
+  );
+}
+
+export function processEvents(
+  items: GCalEvent[],
+  calendarName: string
+): CalendarEvent[] {
+  const events: CalendarEvent[] = [];
+  const changed: ChangedCalendarEvent[] = [];
+  const cancelled: CancelledCalendarEvent[] = [];
+
+  items.forEach(event => {
+    if (event.originalStartTime) {
```

Comment:
Can you combine these two if statements like this?
`if(event.originalStartTime && event.status === 'cancelled') {`

Reply:
I did it this way because we also have to check for `originalStartTime` and `confirmed` also which indicates a changed calendar event.